test/test_helper.exs

```diff
@@ -1,2 +1,3 @@
 Application.ensure_all_started(:stream_data)
+Skate.Migrate.migrate()
```

Comment:
I'd like to do `mix ecto.drop; mix ecto.create; mix ecto.migrate`, but I can't figure out how to do the `drop` or `create` from inside elixir, I can only find command line mix tasks for them.
I could do `Migrator.run(:down, :all)` `Migrator.run(:up, :all)`, which would have a similar effect, but probably be a bit slower.
Do you know how to do the `drop` or `create`, or do you think it's worth it to run all the rollbacks sequentially?

Reply:
You can get at Mix tasks programatically via the `Mix.Tasks` namespace. My preference would be to do the drop/create dance rather than the rollback/migrate dance. Not so much because it'll be faster but because it'll be more robust. But this is the problem I get when I try running the task from IEx:

```
iex(5)> Mix.Tasks.Ecto.Drop.run(["-r", "Skate.Repo"])
** (Mix.Error) The database for Skate.Repo couldn't be dropped: ERROR 55006 (object_in_use) database "skate_dev" is being accessed by other users

There are 10 other sessions using the database.
    (mix 1.10.2) lib/mix.ex:392: Mix.raise/1
    (elixir 1.10.2) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir 1.10.2) lib/enum.ex:783: Enum.each/2
```

...because, of course, we have a thread pool of 10 connections open.

What I'd suggest then is see if there's an easy way to shut all those connections down, run the `Mix.Tasks.Ecto.Drop`, and start new connections. But if it turns into a hassle I don't think it's worth spending a lot of extra time on at this juncture.

Reply:
Even `mix ecto.drop` from the command line fails locally because I have pgAdmin open. I think it's probably not worth it to figure this out.
I also think that doing the full rollback probably isn't worth slowing down the tests, so I'll take that off, too.